packages/kit/src/index.tsx

```diff
@@ -1,37 +1,83 @@
-import { Icon } from './Icon'
-import { QRCode } from './QRCode'
-import { SelectField } from './SelectField'
-import { Slider, SliderField } from './Slider'
-import { SortableList } from './SortableList'
-import { SwitchField } from './SwitchField'
-
-export const RiseComponents = {
-  RiseQRCode: {
-    component: QRCode,
-    validator: QRCode.validate,
-  },
-  RiseIcon: {
-    component: Icon,
-    validator: Icon.validate,
-  },
-  RiseSelectField: {
-    component: SelectField,
-    validator: SelectField.validate,
-  },
-  RiseSlider: {
-    component: Slider,
-    validator: Slider.validate,
-  },
-  RiseSliderField: {
-    component: SliderField,
-    validator: SliderField.validate,
-  },
-  RiseSortableList: {
-    component: SortableList,
-    validator: SortableList.validate,
-  },
-  RiseSwitchField: {
-    component: SwitchField,
-    validator: SwitchField.validate,
-  },
+import { ReactElement } from 'react'
+import z from 'zod'
+
+function createServerComponent<T = any>(type: string, validator: (props: T) => T) {
+  return (props: T): ReactElement => ({
```

Comment:
We can't change return type of JSX Factory, so we have to return shape that satisfies React.Element type. This is an open issue in TypeScript from 2018. This shape is quite good, but we're passing `key: null` since we're not using that field.